.github/workflows/build+test+deploy.yml

```diff
@@ -1,27 +1,18 @@
-name: CI
+name: Build, Test and SelfCheck
 
 on:
-  pull_request:
-  workflow_dispatch:
-
-  # to execute once a day (more info see https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule )
-  schedule:
-    - cron: '0 0 * * *'
-
   push:
-    tags:
-      - 'v*'
+    branches:
+      - master
```

Comment:
@xperiandri I know that you have put '**' for pull_request, but still putting this filter here will mean that a contributor that is about to post a PR won't get CI feedback until she proposes the PR, which is unnecessary. Contributors should be able to see the CI statuses before creating PRs.